otter/integration/lib/autoscale.py

```diff
@@ -132,6 +138,91 @@ def set_launch_config(self, rcs, launch_config):
+
+        :param TestResources rcs: The integration test resources instance.
+            This provides useful information to complete the request, like
+            which endpoint to use to make the API request.
+
+        :return:
+        """
+        # Retrieve the configuration from the stored response to group
+        # creation since that will include non-provided values
+        config = [g['group']['groupConfiguration'] for g in rcs.groups
+                  if g['group']['id'] == self.group_id]
+        # Since this should always provide a match for a valid scaling group,
+        # allow the exception to be raised if nothing is found.
+        if verbosity > 0:
+            print('ScalingGroup.get_group_config will return: ')
+            pp.pprint(config[0])
+        return config[0]
+
+    def replace_group_config(self, rcs, replacement_config):
+        """
+        Replace the current group configuration with the provided config
+        and update the stored groupConfiguration information.
+
+        :param TestResources rcs: The integration test resources instance.
+            This provides useful information to complete the request, like
+            which endpoint to use to make the API request.
+        :param dict replacement_config: A dictionary representation of
+            the JSON description of a scaling group groupConfiguration
+            Note that since this is a replacement config, all fields
+            in the JSON descirptor are mandatory.
+
+        """
+
+        def record_results(_, replacement_config):
+            # Replace the stored value in the group_config
+            self.group_config["groupConfiguration"] = replacement_config
+            # Find the correct group from the list (by id)
+            for g in rcs.groups:
+                if g['group']['id'] == self.group_id:
+                    g['group']['groupConfiguration'] = replacement_config
+            if verbosity > 0:
+                print('Update group_config with {}'.format(replacement_config))
+            return rcs
+
+        return (
+            treq.put(
+                "{0}/groups/{1}/config".format(rcs.endpoints["otter"],
+                                               self.group_id),
+                json.dumps(replacement_config),
+                headers=headers(str(rcs.token)),
+                pool=self.pool,
+            )
+            .addCallback(check_success, [204])
+            .addCallback(record_results, replacement_config)
+        )
+
+    def update_group_config(self, rcs, name=None, cooldown=None,
+                            minEntities=None, maxEntities=None, metadata=None):
+        """
+        Update the group configuration of a scaling group. The provided
+        values will be updated, any others will remain unchanged.
+        """
+        # Get the old config
+        old_config = self.get_group_config(rcs)
+        new_config = {}
+
+        new_config["name"] = name or old_config["name"]
```

Comment:
Apologies I didn't catch this before - might want to just loop over all the values here and check for None.  Since if you want to overwrite the name with "", this would not overwrite since "" evaluates to `False`.  Likewise if you wanted to set `minEntities` or `maxEntities` to 0, or if you wanted to set metadata to `{}`.